### PR TITLE
This is a large contribution of changes from Esgyn

### PR DESCRIPTION
**Pull request by sbroeder**
TransactionManager and libraries that are collectively
much better tested and hardened than Trafodion, but are too
numerous and complex to cherry pick individually.

The changes include:
Dozens of bug fixes
Improved recovery after HBase failures
Better error reporting and tracing, which includes region keys for easier debugging
Reduced UnknownTransactionExceptions resulting from duplicate transaction registration
Better TLOG management including reducing the default number of tables

### CODE REVIEW

**Comment by sbroeder** [comment]
Jenkins extra tests

**Comment by sbroeder** [comment]
jenkins retest

**Comment by svarnau** [comment]
Jenkins is saying you have files without proper license statements:

Unapproved licenses:
2017-04-30 03:03:38 
2017-04-30 03:03:38   trafodion/core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/coprocessor/transactional/generated/SsccRegionProtos.java
2017-04-30 03:03:38   trafodion/core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/coprocessor/transactional/generated/TrxRegionProtos.java

**Comment by sbroeder** [comment]
Jenkins retest

**Comment by sbroeder** [comment]
jenkins extra tests

**Comment by sbroeder** [comment]
Jenkins extra tests

**Comment by DaveBirdsall** [comment]
@sbroeder, thanks for this contribution. Can you recommend knowledgeable reviewers to take a look?

**Comment by sbroeder** [comment]
@shangshengtung @narendragoyal @prashanth-vasudev Please review

**Comment by sbroeder** [comment]
Jenkins extra tests

**Comment by sbroeder** [comment]
Jenkins extra tests

**Comment by sbroeder** [comment]
Jenkins extra retest

**Comment by shangshengtung** [comment]
The change looks good to me.

File: core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionManager.java [hunk 1]
```diff
@@ -2659,19 +2758,37 @@ public void createTable(final TransactionState transactionState, HTableDescripto
...
+            TableName tableName = desc.getTableName();
+            while (keepPolling) {
+
+               try {
+                 Thread.sleep(1000); // sleep one second or until interrupted
+               }
+               catch (InterruptedException e) {
+                  // ignore the interruption and keep going
+               }
+
```
Comment (prashanth-vasudev) [inline core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/transactional/TransactionManager.java]: Generally table is enabled as part of admin coming back from a waited call. If not enabled by any chance,  may be it is better to sleep only if isTableEnabled() call comes back as disabled.  The concern is create table will now sleep for one second always.

**Comment by sbroeder** [comment]
jenkins extra retest

**Comment by sandhyasun** [comment]
Looks good ..merging this in now.

**Pull request merged**